pkg/platform/aws/aws_internal_test.go

```diff
@@ -222,3 +223,64 @@ func TestConfigurationIsValidWhen(t *testing.T) {
 		})
 	}
 }
+
+func Test_config_checkCPUManagerPolicy(t *testing.T) {
+	tests := []struct {
+		name        string
+		WorkerPools []workerPool
+		want        hcl.Diagnostics
+	}{
+		{
+			name: "valid_cpu_manager_policy_values",
+			WorkerPools: []workerPool{
+				{CPUManagerPolicy: ""},
+				{CPUManagerPolicy: "static"},
+				{CPUManagerPolicy: "none"},
+			},
+			want: nil,
+		},
+		{
+			name: "invalid_cpu_manager_policy_value",
+			WorkerPools: []workerPool{
+				{CPUManagerPolicy: ""},
+				{Name: "foobar", CPUManagerPolicy: "foobar"},
+			},
+			want: hcl.Diagnostics{&hcl.Diagnostic{
+				Severity: hcl.DiagError,
+				Summary:  "invalid cpu_manager_policy",
+				Detail:   "Worker pool 'foobar' has invalid cpu_manager_policy 'foobar'",
+			}},
+		},
+		{
+			name: "all_invalid_values",
+			WorkerPools: []workerPool{
+				{Name: "foo", CPUManagerPolicy: "foo"},
+				{Name: "bar", CPUManagerPolicy: "bar"},
+			},
+			want: hcl.Diagnostics{
+				&hcl.Diagnostic{
+					Severity: hcl.DiagError,
+					Summary:  "invalid cpu_manager_policy",
+					Detail:   "Worker pool 'foo' has invalid cpu_manager_policy 'foo'",
+				},
+				&hcl.Diagnostic{
+					Severity: hcl.DiagError,
+					Summary:  "invalid cpu_manager_policy",
+					Detail:   "Worker pool 'bar' has invalid cpu_manager_policy 'bar'",
+				},
+			},
+		},
+	}
+	for _, tt := range tests {
+		tt := tt
+		t.Run(tt.name, func(t *testing.T) {
+			c := &config{
+				WorkerPools: tt.WorkerPools,
+			}
+
+			if got := c.checkCPUManagerPolicy(); !reflect.DeepEqual(got, tt.want) {
```

Comment:
DeepEqual on diagnostic text seems annoying to maintain. Why can't we just use `HasErrors()`?

Also, `cmp.Diff` usually is nicer to use than `DeepEqual` :)